packages/react-component-library/src/components/SelectBase/SelectBaseOption.tsx

```diff
@@ -39,7 +39,9 @@ export const SelectBaseOption = React.forwardRef<
       {...rest}
     >
       {icon}
-      <StyledOptionText>{children}</StyledOptionText>
+      <StyledOptionText title={children.toString()}>
```

Comment:
This doesn't play nice with the icons and badges story unfortunately: 

![image](https://user-images.githubusercontent.com/66470099/152978344-d25c75a9-f669-4757-b71e-0f34465f1e30.png)

Reply:
I've updated this to used value instead of the option children.